linkerd/cache/src/lib.rs

```diff
@@ -9,53 +12,47 @@ use std::{
 };
 use tracing::{debug, trace};
 
-pub mod layer;
-
-pub use self::layer::CacheLayer;
-
 #[derive(Clone)]
-pub struct Cache<T, N>
+pub struct Cache<T, N, S>
 where
     T: Eq + Hash,
-    N: NewService<(T, Handle)>,
 {
-    new_service: N,
-    services: Services<T, N::Service>,
+    inner: N,
+    services: Services<T, S>,
 }
 
-/// A tracker inserted into each inner service that, when dropped, indicates the service may be
-/// removed from the cache.
-#[derive(Clone, Debug)]
-pub struct Handle(Arc<()>);
+/// A tracker inserted into each inner service that, when dropped, indicates the
+/// service may be removed from the cache.
+pub type Handle = Arc<()>;
 
 type Services<T, S> = Arc<RwLock<HashMap<T, (S, Weak<()>)>>>;
 
 // === impl Cache ===
 
-impl<T, N> Cache<T, N>
+impl<T, N> Cache<T, N, N::Service>
 where
     T: Eq + Hash,
-    N: NewService<(T, Handle)>,
+    N: NewService<(Handle, T)>,
```

Comment:
This changed because it's (now) idiomatic for services to put their argument on the first side of a tuple.